taichi/python/export_visual.cpp

```diff
@@ -41,6 +41,38 @@ void export_visual(py::module &m) {
                          img.get_data_size());
            })
       .def("screenshot", &GUI::screenshot)
+      .def("set_widget_value",
+           [](GUI *gui, int wid, float value) {
+             *gui->widget_values.at(wid) = value;
+           })
+      .def("get_widget_value",
+           [](GUI *gui, int wid) -> float {
+             return *gui->widget_values.at(wid);
+           })
+      .def("make_slider",
+           [](GUI *gui, std::string text, float init_value, float minimum,
+              float maximum, float step) {
+             auto val = std::make_unique<float>(init_value);
+             auto val_ptr = val.get();
+             gui->widget_values.push_back(std::move(val));
+             gui->slider(text, *val_ptr, minimum, maximum, step);
+             return gui->widget_values.size() - 1;
+           })
+      .def("make_label",
```

Comment:
I wonder why a label must have a floating value? And there's no way to change its text?

Reply:
Please run my example and use key 'a' and 'd'.

Reply:
I'm not saying the printed text as a whole cannot be changed. Instead, it seems that only the floating value part could change? But now I looked at `Label`, I guess this is a restriction from the GUI system itself..

Reply:
I think people could use `gui.text` for more flexibility, given that the `label` is provided by the legacy C++ part, let's try improve it iapr, instead of itpr.